bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Arena.java

```diff
@@ -735,18 +733,15 @@ public boolean addSpectator(@NotNull Player p, boolean playerBefore, Location st
             p.sendMessage(getMsg(p, Messages.COMMAND_JOIN_SPECTATOR_MSG).replace("%bw_arena%", this.getDisplayName()));
 
             /* update generator holograms for spectators */
-            String iso = Language.getPlayerLanguage(p).getIso();
             for (IGenerator o : getOreGenerators()) {
-                o.updateHolograms(p, iso);
+                o.updateHolograms(p);
             }
             for (ITeam t : getTeams()) {
+                if (!t.isShopSpawned()) continue;
```

Comment:
Wont this be false for playets who (re)join the game later on? meaing the holo wont be visible?

Reply:
Update distance is the distance where the holograms should be updated from. 

There are some pros and cons about making it configurable (which arent much of a problem)

Reply:
No, this variable is saved for the teams, once the shop is spawned, the value is set to true in BedWarsTeam#spawnNpcs()